File: sdks/java/basys.sdk/backends/org/eclipse/basyx/aas/backend/connected/aas/submodelelement/property/ConnectedProperty.java
package org.eclipse.basyx.aas.backend.connected.aas.submodelelement.property;

import org.eclipse.basyx.aas.api.resources.IProperty;
import org.eclipse.basyx.aas.api.resources.PropertyType;
import org.eclipse.basyx.aas.backend.connected.aas.submodelelement.ConnectedDataElement;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.submodelelement.property.Property;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.submodelelement.property.valuetypedef.PropertyValueTypeDefHelper;
import org.eclipse.basyx.vab.core.proxy.VABElementProxy;
/**
 * "Connected" implementation of IProperty
 * @author rajashek
 *
 */
public class ConnectedProperty extends ConnectedDataElement implements IProperty {
	private PropertyType type;

	public ConnectedProperty(PropertyType type, String path, VABElementProxy proxy) {
		super(path, proxy);		
		this.type = type;
	}

	@Override
	public PropertyType getPropertyType() {
		return type;
	}

	@Override
	public void setValue(Object value) {
		getProxy().updateElementValue(constructPath(Property.VALUE), value);
		getProxy().updateElementValue(constructPath(Property.VALUETYPE), PropertyValueTypeDefHelper.fromObject(value).toString());

		
	}

	@Override
	public Object getValue() {
		return	getProxy().readElementValue(constructPath(Property.VALUE));
	}

	@Override
	public void setValueId(Object obj) {
		getProxy().updateElementValue(constructPath(Property.VALUEID), obj);
		
	}

	@Override
	public Object getValueId() {
		return getProxy().readElementValue(constructPath(Property.VALUEID));
	}





}

File: sdks/c++/basys.sdk.cc/src/aas/aas/submodelelement/property/connected/ConnectedProperty.cpp
/*
 * ConnectedProperty.cpp
 *
 *      Author: wendel
 */
#include "ConnectedProperty.h"
#include "basyx/anyTypeChecker.h"

namespace basyx {
namespace aas {
namespace submodelelement {
namespace property {

ConnectedProperty::ConnectedProperty(PropertyType type, std::shared_ptr<vab::core::proxy::IVABElementProxy> proxy) :
  backend::ConnectedDataElement(std::move(proxy)),
  type(type)
{}

PropertyType ConnectedProperty::getPropertyType() const
{
  return this->type;
}

void ConnectedProperty::setValue(const basyx::any & value)
{
  this->getProxy()->updateElementValue(PropertyPaths::VALUE, value);
  this->getProxy()->updateElementValue(PropertyPaths::VALUETYPE, basyx::getPropertyTypeInfo(value));
}

basyx::any ConnectedProperty::getValue() const
{
  return this->getProxy()->readElementValue(PropertyPaths::VALUE);
}

void ConnectedProperty::setValueId(const basyx::any & valueId)
{
  this->getProxy()->updateElementValue(PropertyPaths::VALUEID, valueId);
}

basyx::any ConnectedProperty::getValueId() const
{
  return this->getProxy()->readElementValue(PropertyPaths::VALUEID);
}

void ConnectedProperty::setId(const std::string & id)
{
  ConnectedDataElement::setId(id);
}

std::string ConnectedProperty::getId() const
{
  return ConnectedDataElement::getId();
}

}
}
}
}
